lib/question.js

```diff
@@ -1307,7 +1307,11 @@ module.exports = {
                 locals.iqqrcode = new QR({
                     content: `${config.PLpeekUrl}/${locals.variant.instance_question_id}`,
                 }).svg();
-                callback(null);
+                sqldb.callOneRow('format_date_full_compact', [locals.variant.date, locals.course_instance.display_timezone], (err, result) => {
+                    if (ERR(err, callback)) return;
+                    locals.variant.formatted_date = result.rows[0].format_date_full_compact;
+                    callback(null);
+                });
```

Comment:
All of those functions essentially return a `variants` record, which is complicated when I say "return variants plus this other text column". It seems better to add the `formatted_date` column to the `variants` table and calculate it on `sprocs/variants_insert.sql` and then just use it in the other functions. Does that seem reasonable?

Even if the `formatted_date` column was empty, the structure would be there and we could write things into it on the other select queries? (I assume; I haven't tried that yet.) Or we could do a migration to add it to all of the missing `variants` rows, but presently `sprocs` are loaded after `migrations` in `server.js` so that may take some manual DB updating?)